pkg/api/validate/dynamic/quota.go

```diff
@@ -11,114 +11,64 @@ import (
 	"github.com/Azure/ARO-RP/pkg/api"
 )
 
-func addRequiredResources(requiredResources map[string]int, vmSize api.VMSize, count int) error {
+type azureVM struct {
+	Size      api.VMSize
+	CoreCount int
+	Family    string
+}
+
+func addRequiredResources(requiredResources map[string]int, vmSize api.VMSize, role string, count int) error {
+	var vmTypesMap map[api.VMSize]azureVM = map[api.VMSize]azureVM{
```

Comment:
```suggestion
	vmTypesMap := map[api.VMSize]azureVM{
```

or even this so we don't have to defien a single use struct

```suggestion
	vmTypesMap := map[api.VMSize]struct {
		CoreCount int
		Family    string
	}{
```